### ISSUE TITLE
ASYNC115 False Positive

### ISSUE THREAD

**Issue by rmorshea-atom-computing** [astral-sh/ruff#17728]
### Summary

The [ASYNC115](https://docs.astral.sh/ruff/rules/async-zero-sleep/#async-zero-sleep-async115) rule states that it checks for usages of `trio.sleep(0)` or `anyio.sleep(0)` and replaces them with `trio.lowlevel.checkpoint()`.

This produces false positives in the case where you are using `anyio` with the `asyncio` backend.

I use `anyio` because it adheres to the rules of structured concurrency while remaining compatible with whatever async backend my users choose.

### Version

0.8.6

**Comment by MichaReiser** [comment]
Could you share a code example? I'm not that familiar with async libraries and it's not clear to me what issue you're facing here.

**Comment by rmorshea-atom-computing** [comment]
Here's the sample:

```python
import anyio

async def main():
    await anyio.sleep(0)  # error: ASYNC115
```
